### ISSUE TITLE
No ability to customise the capabilities that Docker keeps

### ISSUE THREAD

**Issue by kdesysadmin** [KostyaSha/yet-another-docker-plugin#148]
As part of our Continuous Integration system it seems that certain unit tests we run require the CAP_SYS_PTRACE capability to be retained, something Docker drops by default. While Docker allows specifying which capabilities to keep, this plugin unfortunately doesn't have such an option.

As we'd like to be able to support these unit tests properly, could the ability to keep certain specified Capabilities please be added? For our purposes using privileged containers isn't workable.

**Comment by KostyaSha** [comment]
https://github.com/KostyaSha/yet-another-docker-plugin/blob/master/yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/commons/DockerCreateContainer.java
+ config.groovy

But it depends on https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/api/command/CreateContainerCmd.java
Seems for jenkins we need also enum

**Comment by kdesysadmin** [comment]
Sorry, i'm not sure what you mean here - is this something which can be scripted using Groovy that i've missed?

**Comment by KostyaSha** [comment]
No, YAD needs additional createcontainer options that available in docker-java

**Comment by kdesysadmin** [comment]
Okay. Will we need to find someone to work on this (ie. do you have plans to implement this at some point)?

**Comment by KostyaSha** [comment]
Sorry, very busy. I plan to go through issues soon and resolve as much as i can.

**Comment by kdesysadmin** [comment]
Not a problem. If we can provide a patch / PR will that make things easier?

**Comment by KostyaSha** [comment]
feel free to popup if you still need it.

**Comment by bcooksley** [comment]
This change is something we are very much still interested in yes.

**Comment by timosachsenberg** [comment]
same problem here. any updates?
